kernel: build.sh: Use wget instead of curl to get kernel tars

dakkshesh07 opened the pull request:
sometimes i ran into this error in some distro(s) when the kernel build script was extracting the kernel tar file:

> xz: (stdin): File format not recognized
> tar: Child returned status 1
> tar: Error is not recoverable: exiting now

after some checking, i found that when the file is downloaded using curl, the file type is shown as a website html file:

$ file linux-5.14.tar.xz
> linux-5.14.tar.xz: HTML document, ASCII text, with very long lines (670)

to fix this, we can use wget instead of curl, after using wget:

$ file linux-5.14.tar.xz
> linux-5.14.tar.xz: XZ compressed data, checksum CRC64

nathanchance commented:
This explanation does not make much sense to me. What did the HTML file say? It probably says something like the file cannot be found, which should happen with `wget` as well. What distros does this happen on? I would rather not change the list of dependencies, especially since we use `curl` to download binutils in `utils.py` and it has been that way since the beginning so people will not expect another dependency, possibly resulting in CI breakage, but if there is really no other way, you need to update the README to add `wget` as a dependency as well.

@msfjarvis thoughts?

dakkshesh07 commented:
> This explanation does not make much sense to me. What did the HTML file say? It probably says something like the file cannot be found, which should happen with `wget` as well. What distros does this happen on? I would rather not change the list of dependencies, especially since we use `curl` to download binutils in `utils.py` and it has been that way since the beginning so people will not expect another dependency, possibly resulting in CI breakage, but if there is really no other way, you need to update the README to add `wget` as a dependency as well.
> 
> @msfjarvis thoughts?

Well, I have tried building with debian based distros like ubuntu, pop os, pure debian, it all worked fine

Few days back I moved to manjaro and I got this issue, to double check I tried building on manjaro + only the packages given in the the readme as well. But same issue

So maybe the issue is there with only arch based distros ig.

dakkshesh07 commented:
Okay, so I tried building the binutils and the same thing happened, it couldn't extract the tar.gz file, I changed that to wget and everything works fine

msfjarvis commented:
> Okay, so I tried building the binutils and the same thing happened, it couldn't extract the tar.xz file, I changed that to wget and everything works fine

We need to see what curl is downloading, like Nathan already asked for in his first reply. We can't accept your solution if you won't let us understand the problem first.

dakkshesh07 commented:
> > Okay, so I tried building the binutils and the same thing happened, it couldn't extract the tar.xz file, I changed that to wget and everything works fine
> 
> We need to see what curl is downloading, like Nathan already asked for in his first reply. We can't accept your solution if you won't let us understand the problem first.

Yeah to get some more info I tried clean building clang again but before that I had deleted and reinstalled curl, and looks like the problem is gone :/

I tried building 2-3 times no issues with curl, so I will close this pr.

Thank you
